src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -1,153 +1,327 @@
     }
 
     /**
      * Return true if at least one value in input matches the predicate
-     * @param input non-null immutable array of ints
+     *
+     * @param input     non-null immutable array of ints
      * @param predicate invoke the predicate.test(int value) on each input element
      */
     @Override
     public boolean someMatch(final int[] input, IntPredicate predicate) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        boolean b = false;
+        for (int i : input) {
+            if (predicate.test(i)) {
+                b = true;
+            }
+        }
+        return b;
     }
 
     /**
      * Return true if all values processed by function, matches the predicate
-     * @param input non-null immutable array of Strings. No element is null
-     * @param function invoke function.applyAsInt(String value) to transform all the input elements into an int value
+     *
+     * @param input     non-null immutable array of Strings. No element is null
+     * @param function  invoke function.applyAsInt(String value) to transform all the input elements into an int value
      * @param predicate invoke predicate.test(int value) to test the int value obtained from the function
      */
     @Override
-    public boolean allMatch(final String[] input,
-                            ToIntFunction<String> function,
-                            IntPredicate predicate) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+    public boolean allMatch(final String[] input, ToIntFunction<String> function, IntPredicate predicate) {
+        int size = input.length;
+        int count = 0;
+        boolean rt = false;
+        for (String s : input) {
+            if (predicate.test(function.applyAsInt(s))) {
+                count++;
+            }
+        }
+        if (count == size) {
+            rt = true;
+        }
+        return rt;
     }
 
     /**
      * Copy values into a separate array from specific index to stopindex
-     * @param input non-null array of ints
+     *
+     * @param input          non-null array of ints
      * @param startInclusive the first index of the element from input to be included in the new array
-     * @param endExclusive the last index prior to which the elements are to be included in the new array
+     * @param endExclusive   the last index prior to which the elements are to be included in the new array
      * @throws IllegalArgumentException when parameters are outside of input index bounds
      */
     @Override
     public int[] copyValues(int[] input, int startInclusive, int endExclusive) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        if (startInclusive < 0 || startInclusive > endExclusive) {
+            throw new IllegalArgumentException("copy value out of bounds");
+        }
+        int arr[] = new int[endExclusive - startInclusive];
```

Comment:
```suggestion
        int[] arr = new int[endExclusive - startInclusive];
```
please use java-style array declaration

Reply:
Done